src/components/Settings.js

```diff
@@ -489,7 +489,14 @@ class SettingsModal extends React.Component {
         var ref_id_value = bibleName + '_' + refLangCodeValue.toLowerCase() + '_' + refVersion.toLowerCase(),
             ref_entry = {},
             ref_arr = [],
-            files = fs.readdirSync(Array.isArray(refFolderPath) ? refFolderPath[0] : refFolderPath);
+            files = [],
+            ref_files = fs.readdirSync(Array.isArray(refFolderPath) ? refFolderPath[0] : refFolderPath);
+           
+            ref_files.map((file)=>{
```

Comment:
Let's use [JS filter method](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) instead. Something like:
```
ref_files = ref_files.filter(file => file.match(/(.*)(\.usfm|\.sfm)/i));
```

You need not create another variable `files` here.

Reply:
oki